gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -99,20 +146,22 @@ def run(
             outlier_weights.append(algorithm.computeOutlierWeights())
 
         # compute average outlier weight
-        avg_outlier_weights = {}
+        avg_outlier_weights = defaultdict(float)
         for outlier_weight_dict in outlier_weights:
-            for index_pair, weight in outlier_weight_dict.items():
-                if index_pair in avg_outlier_weights:
-                    avg_outlier_weights[index_pair] += weight / len(outlier_weights)
-                else:
-                    avg_outlier_weights[index_pair] = weight / len(outlier_weights)
+            for index_pair in valid_i2_i1:
+                avg_outlier_weights[index_pair] += outlier_weight_dict[index_pair]
+
+        for index_pair in avg_outlier_weights:
+            avg_outlier_weights[index_pair] /= len(projection_directions)
```

Comment:
Can we move this logic to a new private function?

Reply:
Im keeping it as is for now, the API for that method will likely change, using valid_i1_i2 is a (hopefully temporary) workaround. Im not able to iterate on some gtsam structures like BinaryMeasurementsUnit3 in the M1 python build.